### DISCUSSION TITLE
File Hook Listener

### DISCUSSION THREAD

**Discussion by genesisbertiz** [strapi/strapi#7960] [Legacy Topics]
Is there a way to get the file information that is going to delete?
I mean a hook that will listen when there's a file will be deleted?
just like **beforeDelete** function.

**Comment by lauriejim** [comment]
Hello! 
Lifecycle functions exist in Strapi - https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks
These functions exist for all models. You have the file generated for your APIs in your application.
To achieve what you are trying to do, you will have to customize the File model lice cycle function.
Because the File model is in a Plugin, you will have to follow this documentation.https://strapi.io/documentation/v3.x/concepts/customization.html#plugin-extensions
